packages/compass-aggregations/src/components/pipeline-builder-workspace/pipeline-as-text-workspace/pipeline-editor.tsx

```diff
@@ -107,6 +130,30 @@ export const PipelineEditor: React.FunctionComponent<PipelineEditorProps> = ({
 
   return (
     <div className={containerStyles} data-testid="pipeline-as-text-editor">
+      <div className={editorContainerActionsStyles}>
+        <Button
+          size="xsmall"
+          aria-label="Copy pipeline"
+          title="Copy pipeline (Ctrl+Shift+C)"
+          onClick={() => {
+            editorRef.current?.execCommand('copy-all');
+          }}
+          className={editorContainerActionButtonStyles}
+        >
+          <Icon title={null} glyph="Copy"></Icon>
+        </Button>
+        <Button
+          size="xsmall"
+          aria-label="Format pipeline"
+          title="Format pipeline (Ctrl+Shift+B)"
+          onClick={() => {
+            editorRef.current?.execCommand('prettify');
+          }}
+          className={editorContainerActionButtonStyles}
+        >
+          <Icon title={null} glyph="Checkmark"></Icon>
```

Comment:
That's the icon used in the new designs for the code editor, I am not sure it's the best one to communicate code formatting, but there is noting else in leafygreen anyway

Reply:
The commands added in this PR and switching logic to use prettier will be already helpful for the implementation that we will need to do for the rebranded version of the editor, so it's not wasted effort

Reply:
> here even before we get to the new editor visual brand tickets.

What I'm missing is why would that not work time-wise. Isn't the editor component part of .. at worst the next GA when we would also enable the FF for the editing pipeline as text?

Reply:
So far seems like visual brand updates can take quite some time to roll out, so FWIW we might be able to roll this out faster taking into account that it's only telemetry we are missing in the scope of this project at the moment and there are at least 2-3 decently sized tickets for adding this functionality in all editors. The only thing here that we will "throw away" is adding these two buttons explicitly, which is very little in terms of effort spent, commands are already added inside the editor package. As we have currently time planned for this project and the ticket was in the current sprint, I did it right now including the buttons part in this component

Reply:
Let's talk about it separately. If you say that this change is this much cheaper and good enough and that implementing the new designs would drag too long compared to what we planned maybe we should consider the opportunity of making this our plan A for editors (especially if there is a quick way to fit a language switcher too), and defer the new designs for when we would have a stronger need.

Reply:
I don't have a strong opinion: that's what we decided on before and so I did it. If you have a preference to not do it now I am happy to close this PR, exclude this from project scope, and restore these changes later when we will be eventually working on relevant parts of the editor. I don't think that these projects overlap enough to include all the work needed to finish the brand update for editor in the scope of the project we almost finished

Reply:
Actually I. was saying the opposite, we should propose to make this the "standard" design for those editors since after all .. it looks good and cost seems to be different :)

Reply:
We will have to use leafygreen code editors in other places (for read-only for example, like export-to-language) anyway though, right? And then we have two different editor designs in the app which is something we were trying to avoid, or at least that was my understanding of why we planned editor project like that. This seems like discussion that will require some time and the only reason we included this ticket was because it was something we thought would be fast to do (it was implementation-wise), I'll keep it out of the pipeline project for now, we can decide to go with this design in the editor project if we really think it's feasible, I'd prefer this not affect the pipeline editor in any case